cmake/modules/Buildpmem.cmake

```diff
@@ -0,0 +1,53 @@
+function(build_pmem)
+  include(ExternalProject)
+  set(PMDK_ROOT "${CMAKE_BINARY_DIR}/src/nvml")
+  set(PMDK_SRC "${PMDK_ROOT}/src")
+  set(PMDK_INCLUDE "${PMDK_SRC}/include")
+
+  # Use debug PMDK libs in debug lib/rbd builds
+  if(CMAKE_BUILD_TYPE STREQUAL Debug)
+    set(PMDK_LIB_DIR "debug")
+  else()
+    set(PMDK_LIB_DIR "nondebug")
+  endif()
+  set(PMDK_LIB "${PMDK_SRC}/${PMDK_LIB_DIR}")
+
+  ExternalProject_Add(nvml_ext
+      DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+      GIT_REPOSITORY "https://github.com/ceph/nvml.git"
+      GIT_TAG "dd622819dd4ee97d3920f913c70be"
+      SOURCE_DIR ${CMAKE_BINARY_DIR}/src/nvml
+      CONFIGURE_COMMAND ""
+      # Explicitly built w/o NDCTL, otherwise if ndtcl is present on the
+      # build system tests statically linking to librbd (which uses
+      # libpmemobj) will not link (because we don't build the ndctl
+      # static library here).
+      BUILD_COMMAND $(MAKE) NDCTL_ENABLE=n
+      BUILD_IN_SOURCE 1
+      BUILD_BYPRODUCTS "${PMDK_LIB}/libpmem.a"
+      BUILD_BYPRODUCTS "${PMDK_LIB}/libpmemobj.a"
+      INSTALL_COMMAND "true")
+    ExternalProject_Add_Step(nvml_ext forcebuild
```

Comment:
I added "BUILD_ALWAY 1", and removed ExternalProject_Add_Step.

Reply:
> why can't we trust the build stamp file

@lixiaoy1 could you help me understand this?

Reply:
I am afraid that nvml won't build if I update files in nvml for debug, so I added "BUILD_ALWAYS 1". But I retested it just now, it can be built without "BUILD_ALWAYS 1".